### ISSUE TITLE
Starting point for customization 

### ISSUE THREAD

**Issue by alkabal** [UnfinishedBusiness/Xmotion#7]
Hi

I like to start learning about your source code my need I make a lathe version from your good GUI

If you can point me in the good direction this can be very helpful

Br

**Comment by UnfinishedBusiness** [comment]
@alkabal - Take a close look at https://docs.littlevgl.com/ . This is the library that's being used to render the GUI to the framebuffer. It's a phenomenal library.

I also would like to point out for you and anyone else that likes our GUI, we have paused development over speed concerns of the embedded computer (In our case is similar hardware to the Beagle Bone Black which is built into our control circuit board). The GUI is in use on a few customers machines that enjoying being our beta testers and each is happy with the setup even though the software is lacking in some ways. I've personally done as much development as I can to make the GUI look and feel good and fast without hardware graphics acceleration but we don't believe we will be able to build the software out to our goal of having CAD & CAM built into the GUI.

To negate the graphics speed issues we have decided further develop our motion control board to have embedded hardware which has only one job of handling the motion control and GCode interpretation. Then there will be a USB interface with serial communication to the embedded motion controller that can be plugged into any USB host computer. This cuts out MachineKit/LinuxCNC (Which is phenomenal software but can be very hard to configure and troubleshoot). We're calling this our Gen3 control. Our HMI/GUI will look and feel much like the current stable branch GUI here but is Java-based for cross-platform compatibility. This gives us much more flexibility in the computer hardware required and we hope we can move back to having the embedded computer on-boared eventually if the graphics processing power gets better in the future... (I think the technology is there now but TI doesn't seem to want to provide a driver for it for some reason). I hope that anyone that likes this software now will continue to use it and develop it, feel free to fork it. Soon it will be moved into "Stable_Gen2_Branch" and master will be filled with our Java-based HMI
